definitions/checks/foreman/check_external_db_evr_permissions.rb

```diff
@@ -0,0 +1,63 @@
+module Checks
+  module Foreman
+    class CheckExternalDbEvrPermissions < ForemanMaintain::Check
+      metadata do
+        label :external_db_evr_permissions
+        for_feature :foreman_database
+        description 'Check that external DBs have proper EVR extension permissions'
+        tags :pre_upgrade
+        confine do
+          feature(:foreman_database) && !feature(:foreman_database).local? && feature(:katello)
+        end
+      end
+
+      def run
+        return true unless evr_exists?
+
+        error_msg = 'The evr extension is not owned by the foreman DB owner. Please run the ' \
+              'following command on the external foreman database to fix it: ' \
+              'UPDATE pg_extension SET extowner = (SELECT oid FROM pg_authid WHERE ' \
+              "rolname='#{foreman_db_user}') WHERE extname='evr';"
+        fail!(error_msg) unless foreman_owns_evr?
+      end
+
+      private
+
+      def foreman_db_user
+        feature(:foreman_database).configuration['username'] || 'foreman'
+      end
+
+      def evr_exists?
+        evr_exists = feature(:foreman_database).query(query_for_evr_existence)
+        unless evr_exists.empty?
+          return evr_exists.first['evr_exists'] == '1'
+        end
+        return false
```

Comment:
This is not a required change, more food for thought. You could reduce this down to two lines:

```
return false if evr_exists.empty?
return evr_exists.first['evr_exists'] == '1'
```

Reply:
works for me!